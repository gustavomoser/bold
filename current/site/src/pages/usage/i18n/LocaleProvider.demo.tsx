import { LocaleContext, Paginator } from 'bold-ui'
import ptBr from 'bold-ui/i18n/locales/pt-BR'
import React from 'react'

function LocaleProviderDemo() {
  // Using Paginator as example since it uses strings from locale context internally
  return (
    <LocaleContext.Provider value={ptBr}>
      <Paginator page={5} total={10} />
    </LocaleContext.Provider>
  )
}

export default LocaleProviderDemo
