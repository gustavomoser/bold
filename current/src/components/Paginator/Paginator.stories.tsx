import { action } from '@storybook/addon-actions'
import { number } from '@storybook/addon-knobs'
import React from 'react'

import { Paginator } from './Paginator'

export default {
  title: 'Components/Paginator',
}

export const Default = () => (
  <Paginator page={number('page', 4)} total={number('total', 20)} onChange={action('page-changed')} />
)
