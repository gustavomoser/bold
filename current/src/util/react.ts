import { MutableRefObject, Ref } from 'react'

/**
 * Compose multiple react refs.
 *
 * @returns A new ref that passes down the Element to all ref arguments.
 */
export function composeRefs<T>(...refs: Array<Ref<T>>) {
  return (value: T) => refs.forEach(ref => setRef(ref, value))
}

/**
 * Changes the value of a ref.
 * The ref can be a function ref or a object ref.
 *
 * @param ref The ref to have its value changed.
 * @param value The new value.
 */
export function setRef<T>(ref: Ref<T>, value: any) {
  if (typeof ref === 'function') {
    ref(value)
  } else if (ref) {
    const mutableRef = ref as MutableRefObject<T>
    mutableRef.current = value
  }
}

/**
 * Compose multiple event handlers.
 *
 * @returns A new function that triggers all handlers passed as parameters, in order.
 */
export function composeHandlers(...handlers: Array<(...params: any) => void>) {
  return (...args: any[]) => handlers.forEach(handler => handler && handler(...args))
}

/**
 * Invokes the native `value` property setter of an element.
 * From https://github.com/facebook/react/issues/10135#issuecomment-401496776
 *
 * @param element The element to invoke the setter on.
 * @param value The value to be set.
 */
export function setNativeValue(element: HTMLElement, value: any) {
  const prototype = Object.getPrototypeOf(element)
  const { set: prototypeValueSetter = null } = Object.getOwnPropertyDescriptor(prototype, 'value') || {}
  const { set: valueSetter = null } = Object.getOwnPropertyDescriptor(element, 'value') || {}

  if (prototypeValueSetter && valueSetter !== prototypeValueSetter) {
    prototypeValueSetter.call(element, value)
  } else if (valueSetter) {
    valueSetter.call(element, value)
  } else {
    throw new Error('The given element does not have a value setter')
  }
}
