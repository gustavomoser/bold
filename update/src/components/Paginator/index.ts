export { Paginator } from './Paginator'
export type { PaginatorProps } from './Paginator'
