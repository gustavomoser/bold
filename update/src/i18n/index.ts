export { LocaleContext, useLocale } from './LocaleContext'
export type { LocaleConfiguration } from './LocaleContext'
