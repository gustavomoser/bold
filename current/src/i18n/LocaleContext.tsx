import { createContext, useContext } from 'react'

import defaultLocale from './locales/en-US'

export type LocaleConfiguration = typeof defaultLocale

export const LocaleContext = createContext<LocaleConfiguration>(defaultLocale)

export function useLocale() {
  return useContext(LocaleContext)
}
