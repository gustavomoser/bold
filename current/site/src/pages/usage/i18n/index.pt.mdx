import { Demo } from '../../../components/Demo'

# Internacionalização (i18n)

Os componentes do Bold usam por padrão a linguagem do browser para determinar o _locale_.

É possível customizar os textos usados pelos componentes usando `LocaleContext.Provider` e acessá-las através da _hook_ `useLocale`.

A hook `useLocale` retorna um mapa de palavras usadas internamente pelos componentes do bold, como o Calendar, Paginator etc.

<Demo src='pages/usage/i18n/LocaleProvider' />

## _Locales_ disponíveis

Verifique os _locales_ disponíveis na pasta `src/i18n/locales`. Sinta-se livre para abrir um _pull request_ caso seu _locale_ não esteja presente.
