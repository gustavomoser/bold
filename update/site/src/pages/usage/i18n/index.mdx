import { Demo } from '../../../components/Demo'

# Internationalization (i18n)

The components use the user browser language to determine the locale.

You can customize the strings used by the components using the `LocaleContent.Provider` component and access them using the `useLocale` hook.

The `useLocale` hook returns a map of strings used internally by some bold components, like Calendar, Paginator etc.

<Demo src='pages/usage/i18n/LocaleProvider' />

## Available locales

Check the available locales on the `src/i18n/locales` folder. Feel free to open a pull request if your locale is not there yet.
