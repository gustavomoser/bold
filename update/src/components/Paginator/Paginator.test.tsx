import { fireEvent, render } from '@testing-library/react'
import React from 'react'

import { LocaleContext } from '../../i18n'
import ptBr from '../../i18n/locales/pt-BR'

import { Paginator } from './Paginator'

it('should render correctly', () => {
  expect(render(<Paginator page={5} total={10} />).container).toMatchSnapshot()
  expect(render(<Paginator page={0} total={10} />).container).toMatchSnapshot()
  expect(render(<Paginator page={0} total={9} />).container).toMatchSnapshot()
})

it('should call onChange with correct values', () => {
  const handleChange = jest.fn()
  const { container } = render(<Paginator page={4} total={10} onChange={handleChange} />)
  const input = container.querySelector('input')

  expect(handleChange).not.toHaveBeenCalled()

  fireEvent.change(input, { target: { value: '4' } })
  fireEvent.blur(input)
  expect(handleChange).toHaveBeenLastCalledWith(3)

  fireEvent.change(input, { target: { value: '10' } })
  fireEvent.blur(input)
  expect(handleChange).toHaveBeenLastCalledWith(9)

  fireEvent.change(input, { target: { value: '11' } })
  fireEvent.blur(input)
  expect(handleChange).toHaveBeenLastCalledWith(9)

  fireEvent.change(input, { target: { value: '0' } })
  fireEvent.blur(input)
  expect(handleChange).toHaveBeenLastCalledWith(9)

  fireEvent.change(input, { target: { value: '1' } })
  fireEvent.blur(input)
  expect(handleChange).toHaveBeenLastCalledWith(0)
})

it('should call onChange when Enter is pressed', () => {
  const handleChange = jest.fn()
  const { container } = render(<Paginator page={4} total={10} onChange={handleChange} />)
  const input = container.querySelector('input')

  fireEvent.change(input, { target: { value: '8' } })
  fireEvent.keyDown(input, { key: 'Enter' })

  expect(handleChange).toHaveBeenLastCalledWith(7)
  expect(handleChange).toHaveBeenCalledTimes(1)
})

it('should reset value if value is not an integer', () => {
  const handleChange = jest.fn()
  const { container } = render(<Paginator page={4} total={10} onChange={handleChange} />)
  const input = container.querySelector('input')

  fireEvent.change(input, { target: { value: 'a' } })
  fireEvent.blur(input)

  expect(input.value).toBe('5')
})

it('should reset value if value is less than 1 or bigger than total', () => {
  const handleChange = jest.fn()
  const { container } = render(<Paginator page={4} total={10} onChange={handleChange} />)
  const input = container.querySelector('input')

  fireEvent.change(input, { target: { value: '0' } })
  fireEvent.blur(input)
  expect(input.value).toBe('5')

  fireEvent.change(input, { target: { value: '11' } })
  fireEvent.blur(input)
  expect(input.value).toBe('5')
})

it('should allow message customization via locale context', () => {
  const { container, queryByText } = render(
    <LocaleContext.Provider value={ptBr}>
      <Paginator page={5} total={10} />
    </LocaleContext.Provider>
  )
  expect(container.querySelectorAll('button')[0].getAttribute('title')).toEqual(ptBr.paginator.previousPage)
  expect(container.querySelector('input').getAttribute('title')).toEqual(ptBr.paginator.currentPage)
  expect(queryByText(`${ptBr.paginator.of} 10`)).toBeTruthy()
  expect(container.querySelectorAll('button')[1].getAttribute('title')).toEqual(ptBr.paginator.nextPage)
})
