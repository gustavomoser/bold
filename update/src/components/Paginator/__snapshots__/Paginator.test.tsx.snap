// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
}

.emotion-1 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
}

.emotion-1>span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0, 0, 0, 0.1);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-1:not(:disabled):hover {
  background-color: rgba(58, 58, 71, 0.1);
}

.emotion-2 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-3 {
  position: relative;
}

.emotion-4 {
  background-color: #FFFFFF;
  border: solid 1px #8F8FA2;
  border-radius: 2px;
  color: #24252E;
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1rem;
  margin: 0;
  padding: calc(0.5rem - 1px) 0.5rem;
  width: 100%;
  transition-property: box-shadow;
  transition-duration: .2s;
  width: 68px;
  text-align: center;
  margin: 0 0.5rem 0 0.25rem;
}

.emotion-4[type="search"]::-webkit-search-decoration {
  display: none;
}

.emotion-4[type="search"]::-webkit-search-cancel-button {
  display: none;
}

.emotion-4[type="search"]::-webkit-search-results-button {
  display: none;
}

.emotion-4[type="search"]::-webkit-search-results-decoration {
  display: none;
}

.emotion-4:required {
  box-shadow: none;
}

.emotion-4::-webkit-input-placeholder {
  color: #696979;
}

.emotion-4::-moz-placeholder {
  color: #696979;
}

.emotion-4:-ms-input-placeholder {
  color: #696979;
}

.emotion-4::placeholder {
  color: #696979;
}

.emotion-4:disabled {
  border-color: #D3D4DD;
  background-color: #F0F0F5;
}

.emotion-4:not(:disabled):hover {
  border-color: #7A7B8E;
}

.emotion-4:not(:disabled):focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-4:not(:disabled):active {
  border-color: #0069D0;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.09);
}

.emotion-5 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5;
  color: #24252E;
}

.emotion-6 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  margin-left: 0.25rem;
}

.emotion-6>span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-6:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0, 0, 0, 0.1);
}

.emotion-6:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-6:not(:disabled):hover {
  background-color: rgba(58, 58, 71, 0.1);
}

<div>
  <div
    class="emotion-0"
  >
    <button
      class="emotion-1"
      title="Previous page"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="emotion-2"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            clip-rule="evenodd"
            d="M14.707 16.294a1 1 0 11-1.414 1.414l-5-5a1 1 0 010-1.414l5.001-5.001a1 1 0 011.414 1.414l-4.294 4.294 4.293 4.293z"
          />
        </svg>
      </span>
    </button>
    <div
      class="emotion-3"
    >
      <input
        class="emotion-4"
        maxlength="4"
        title="Current page"
        type="text"
        value="6"
      />
    </div>
    <span
      class="emotion-5"
    >
      of
       
      10
    </span>
    <button
      class="emotion-6"
      title="Next page"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="emotion-2"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            clip-rule="evenodd"
            d="M8.294 7.707a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5.001a1 1 0 01-1.415-1.414L12.587 12 8.294 7.707z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`should render correctly 2`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
}

.emotion-1 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-1>span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0, 0, 0, 0.1);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-1:not(:disabled):hover {
  background-color: rgba(58, 58, 71, 0.1);
}

.emotion-2 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-3 {
  position: relative;
}

.emotion-4 {
  background-color: #FFFFFF;
  border: solid 1px #8F8FA2;
  border-radius: 2px;
  color: #24252E;
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1rem;
  margin: 0;
  padding: calc(0.5rem - 1px) 0.5rem;
  width: 100%;
  transition-property: box-shadow;
  transition-duration: .2s;
  width: 68px;
  text-align: center;
  margin: 0 0.5rem 0 0.25rem;
}

.emotion-4[type="search"]::-webkit-search-decoration {
  display: none;
}

.emotion-4[type="search"]::-webkit-search-cancel-button {
  display: none;
}

.emotion-4[type="search"]::-webkit-search-results-button {
  display: none;
}

.emotion-4[type="search"]::-webkit-search-results-decoration {
  display: none;
}

.emotion-4:required {
  box-shadow: none;
}

.emotion-4::-webkit-input-placeholder {
  color: #696979;
}

.emotion-4::-moz-placeholder {
  color: #696979;
}

.emotion-4:-ms-input-placeholder {
  color: #696979;
}

.emotion-4::placeholder {
  color: #696979;
}

.emotion-4:disabled {
  border-color: #D3D4DD;
  background-color: #F0F0F5;
}

.emotion-4:not(:disabled):hover {
  border-color: #7A7B8E;
}

.emotion-4:not(:disabled):focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-4:not(:disabled):active {
  border-color: #0069D0;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.09);
}

.emotion-5 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5;
  color: #24252E;
}

.emotion-6 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  margin-left: 0.25rem;
}

.emotion-6>span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-6:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0, 0, 0, 0.1);
}

.emotion-6:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-6:not(:disabled):hover {
  background-color: rgba(58, 58, 71, 0.1);
}

<div>
  <div
    class="emotion-0"
  >
    <button
      class="emotion-1"
      disabled=""
      title="Previous page"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="emotion-2"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            clip-rule="evenodd"
            d="M14.707 16.294a1 1 0 11-1.414 1.414l-5-5a1 1 0 010-1.414l5.001-5.001a1 1 0 011.414 1.414l-4.294 4.294 4.293 4.293z"
          />
        </svg>
      </span>
    </button>
    <div
      class="emotion-3"
    >
      <input
        class="emotion-4"
        maxlength="4"
        title="Current page"
        type="text"
        value="1"
      />
    </div>
    <span
      class="emotion-5"
    >
      of
       
      10
    </span>
    <button
      class="emotion-6"
      title="Next page"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="emotion-2"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            clip-rule="evenodd"
            d="M8.294 7.707a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5.001a1 1 0 01-1.415-1.414L12.587 12 8.294 7.707z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`should render correctly 3`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
}

.emotion-1 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-1>span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0, 0, 0, 0.1);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-1:not(:disabled):hover {
  background-color: rgba(58, 58, 71, 0.1);
}

.emotion-2 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-3 {
  position: relative;
}

.emotion-4 {
  background-color: #FFFFFF;
  border: solid 1px #8F8FA2;
  border-radius: 2px;
  color: #24252E;
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1rem;
  margin: 0;
  padding: calc(0.5rem - 1px) 0.5rem;
  width: 100%;
  transition-property: box-shadow;
  transition-duration: .2s;
  width: 68px;
  text-align: center;
  margin: 0 0.5rem 0 0.25rem;
}

.emotion-4[type="search"]::-webkit-search-decoration {
  display: none;
}

.emotion-4[type="search"]::-webkit-search-cancel-button {
  display: none;
}

.emotion-4[type="search"]::-webkit-search-results-button {
  display: none;
}

.emotion-4[type="search"]::-webkit-search-results-decoration {
  display: none;
}

.emotion-4:required {
  box-shadow: none;
}

.emotion-4::-webkit-input-placeholder {
  color: #696979;
}

.emotion-4::-moz-placeholder {
  color: #696979;
}

.emotion-4:-ms-input-placeholder {
  color: #696979;
}

.emotion-4::placeholder {
  color: #696979;
}

.emotion-4:disabled {
  border-color: #D3D4DD;
  background-color: #F0F0F5;
}

.emotion-4:not(:disabled):hover {
  border-color: #7A7B8E;
}

.emotion-4:not(:disabled):focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-4:not(:disabled):active {
  border-color: #0069D0;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.09);
}

.emotion-5 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5;
  color: #24252E;
}

.emotion-6 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  margin-left: 0.25rem;
}

.emotion-6>span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-6:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0, 0, 0, 0.1);
}

.emotion-6:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-6:not(:disabled):hover {
  background-color: rgba(58, 58, 71, 0.1);
}

<div>
  <div
    class="emotion-0"
  >
    <button
      class="emotion-1"
      disabled=""
      title="Previous page"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="emotion-2"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            clip-rule="evenodd"
            d="M14.707 16.294a1 1 0 11-1.414 1.414l-5-5a1 1 0 010-1.414l5.001-5.001a1 1 0 011.414 1.414l-4.294 4.294 4.293 4.293z"
          />
        </svg>
      </span>
    </button>
    <div
      class="emotion-3"
    >
      <input
        class="emotion-4"
        maxlength="4"
        title="Current page"
        type="text"
        value="1"
      />
    </div>
    <span
      class="emotion-5"
    >
      of
       
      9
    </span>
    <button
      class="emotion-6"
      title="Next page"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="emotion-2"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            clip-rule="evenodd"
            d="M8.294 7.707a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5.001a1 1 0 01-1.415-1.414L12.587 12 8.294 7.707z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
